server/app/com/xsn/explorer/services/synchronizer/MasternodeSynchronizerActor.scala

```diff
@@ -0,0 +1,41 @@
+package com.xsn.explorer.services.synchronizer
+
+import akka.actor.{Actor, ActorRef, ActorSystem, Props}
+import com.xsn.explorer.models._
+import com.xsn.explorer.models.values.IPAddress
+
+class MasternodeSynchronizerActor extends Actor {
+  import context._
+
+  def receive: Receive = {
+    val initialMasternodes = List[rpc.Masternode]()
+
+    behavior(initialMasternodes)
+  }
+
+  private def behavior(masternodes: List[rpc.Masternode]): Receive = {
+    case MasternodeSynchronizerActor.UpdateMasternodes(newMasternodes) =>
+      become(behavior(newMasternodes))
+    case MasternodeSynchronizerActor.GetMasternode(ipAddress) =>
+      sender() ! masternodes.find(x => x.ip.split(":").headOption.contains(ipAddress.string))
+    case MasternodeSynchronizerActor.GetMasternodes =>
+      sender() ! masternodes
+  }
+}
+
+object MasternodeSynchronizerActor {
+  final class Ref private (val ref: ActorRef) extends AnyVal
+
+  object Ref {
+
+    def apply(name: String = "masternode_synchronizer")(implicit system: ActorSystem): Ref = {
```

Comment:
Use `masternode-synchronizer` instead, which is the common format for actor paths, they are like urls